src/combinatorics/simple_graph/matching.lean

```diff
@@ -18,28 +18,46 @@ import combinatorics.simple_graph.basic
 * a `perfect_matching` on a simple graph is a matching in which every
    vertex belongs to an edge.
 
-## Implementation notes
-
-
-TODO: perfect matchings, Hall's Marriage Theorem
+TODO:
+  - Lemma stating that the existence of a perfect matching on `G` implies that
+    the cardinality of `V` is even (assuming it's finite)
+  - Hall's Marriage Theorem
+  - Tutte's Theorem
 
+TODO: Tutte and Hall require a definition of subgraphs.
 -/
 open finset
 universe u
 
 namespace simple_graph
-variables {V : Type u} (G : simple_graph V)
+variables {V : Type u} (G : simple_graph V) (M : set (sym2 V)) (h : M ⊆ edge_set G)
+
+instance : inhabited (set (sym2 V)) :=
+  { default := (complete_graph V).edge_set }
```

Comment:
Is this part actually useful at all? If so, should it go in `basic.lean`?

Reply:
There's already an inhabited instance for `set α` called `set.inhabited` that uses the empty set, so this probably should be removed.  (It seems fine just saying `(complete_graph V).edge_set` when you need it.)